src/common/SoundInput.c

```diff
@@ -3080,15 +3081,10 @@ BOOL Decode4FSKID(UCHAR bytFrameType, char * strCallID, size_t strCallIDLen, cha
 
 	memcpy(bytCall, bytFrameData1, COMP_SIZE);
 	DeCompressCallsign(bytCall, strCallID, strCallIDLen);
-	memcpy(bytCall, &bytFrameData1[COMP_SIZE], COMP_SIZE);
-	DeCompressGridSquare(bytCall, temp);
-
-	if (strlen(temp) > 5)
-	{
-		temp[4] = tolower(temp[4]);
-		temp[5] = tolower(temp[5]);
+	locator_err e = locator_from_bytes(&bytFrameData1[COMP_SIZE], grid);
+	if (e) {
+		ZF_LOGD("Failed to decode grid ID Frame (sender: %s) grid square: %s", strCallID, locator_strerror(e));
```

Comment:
As commented above, this writes an error message following "grid square: ".  So, I'd like to see the invalid decoded grid string written to the debug log.